Tools/SandboxTest.ps1

```diff
@@ -0,0 +1,298 @@
+
+New-Item $tempFolder -ItemType Directory -ErrorAction SilentlyContinue | Out-Null
+
+# Set dependencies
+
+$apiLatestUrl = 'https://api.github.com/repos/microsoft/winget-cli/releases/latest'
+
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+$WebClient = New-Object System.Net.WebClient
+
+function Get-LatestUrl {
+  ((Invoke-WebRequest $apiLatestUrl -UseBasicParsing | ConvertFrom-Json).assets | Where-Object { $_.name -match '^Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.appxbundle$' }).browser_download_url
+}
+
+function Get-LatestHash {
+  $shaUrl = ((Invoke-WebRequest $apiLatestUrl -UseBasicParsing | ConvertFrom-Json).assets | Where-Object { $_.name -match '^Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.txt$' }).browser_download_url
+
+  $shaFile = Join-Path -Path $tempFolder -ChildPath 'Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.txt'
+  $WebClient.DownloadFile($shaUrl, $shaFile)
+
+  Get-Content $shaFile
+}
+
+# Hide the progress bar of Invoke-WebRequest
+$oldProgressPreference = $ProgressPreference
+$ProgressPreference = 'SilentlyContinue'
+
+$desktopAppInstaller = @{
+  fileName = 'Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.appxbundle'
+  url      = $(Get-LatestUrl)
+  hash     = $(Get-LatestHash)
+}
+
+$ProgressPreference = $oldProgressPreference
+
+$vcLibsUwp = @{
+  fileName = 'Microsoft.VCLibs.x64.14.00.Desktop.appx'
+  url      = 'https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx'
+  hash     = '6602159c341bafea747d0edf15669ac72df8817299fbfaa90469909e06794256'
+}
+
+$dependencies = @($desktopAppInstaller, $vcLibsUwp)
+
+# Clean temp directory
+
+Get-ChildItem $tempFolder -Recurse -Exclude $dependencies.fileName | Remove-Item -Force -Recurse
+
+if (-Not [String]::IsNullOrWhiteSpace($Manifest)) {
+  Copy-Item -Path $Manifest -Recurse -Destination $tempFolder
+}
+
+# Download dependencies
+
+Write-Host '--> Checking dependencies'
+
+$desktopInSandbox = 'C:\Users\WDAGUtilityAccount\Desktop'
+
+foreach ($dependency in $dependencies) {
+  $dependency.file = Join-Path -Path $tempFolder -ChildPath $dependency.fileName
+  $dependency.pathInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $dependency.fileName)
+
+  # Only download if the file does not exist, or its hash does not match.
+  if (-Not ((Test-Path -Path $dependency.file -PathType Leaf) -And $dependency.hash -eq $(get-filehash $dependency.file).Hash)) {
+    Write-Host @"
+    - Downloading:
+      $($dependency.url)
+"@
+
+    try {
+      $WebClient.DownloadFile($dependency.url, $dependency.file)
+    }
+    catch {
+      throw "Error downloading $($dependency.url)."
+    }
+    if (-not ($dependency.hash -eq $(get-filehash $dependency.file).Hash)) {
+      throw 'Hashes do not match, try gain.'
+    }
+  }
+}
+
+Write-Host
+
+# Create Bootstrap script
+
+# See: https://stackoverflow.com/a/14382047/12156188

```

Comment:
I'd rename this for a little more clarity. Maybe Update-EnvironmentVariables?

Reply:
Also, this implementation seems a little less fraught to me? https://stackoverflow.com/a/22670892/892770

Reply:
> Update-EnvironmentVariables

Well, it's really a long string, especially if you consider that it will be typed manually. But that's how PowerShell is, I agree with this change.

> Also, this implementation seems a little less fraught to me? https://stackoverflow.com/a/22670892/892770

Definitely. I don't know why I picked the first one, but I'll try and if it works I'll switch.

Reply:
> Update-EnvironmentVariables

Well, it's really a long string, especially if you consider that it will be typed manually. But that's how PowerShell is, I agree with this change.

> Also, this implementation seems a little less fraught to me? https://stackoverflow.com/a/22670892/892770

Definitely. I don't know why I picked the first one, but I'll try and if it works I'll switch.